src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,394 @@
+        } // We will need conversion here on BC 1.54
+        return null;
+    }
+
+    /**
+     * Obtain {@link PublicKey} object from the read PEM. If the PEM data contained other object type like
+     * {@link KeyPair} or {@link Certificate}, the public key will be extracted from them.
+     * 
+     * @return {@link PublicKey} with the public key, null if a public key could not be obtained from the current data
+     */
+    @Nullable
+    public PublicKey getPublicKey() {
+        if (pemObject instanceof PublicKey) {
+            return (PublicKey) pemObject;
+        } else if (pemObject instanceof KeyPair) {
+            return ((KeyPair) pemObject).getPublic();
+        } else if (pemObject instanceof Certificate) {
+            return ((Certificate) pemObject).getPublicKey();
+        }
+        return null;
+    }
+
+    /**
+     * Obtain {@link Certificate} object from the read PEM.
+     * 
+     * @return {@link Certificate} with the certificate, null if a certificate could not be obtained from the current
+     * data
+     */
+    @Nullable
+    public Certificate getCertificate() {
+        if (pemObject instanceof Certificate) {
+            return ((Certificate) pemObject);
+        }
+        return null;
+    }
+
+    /**
+     * Obtain {@link PrivateKey} object from the read PEM. If the PEM data contained other object type like
+     * {@link KeyPair}, the private key will be extracted from them.
+     * 
+     * @return {@link PrivateKey} with the private key, null if a private key could not be obtained from the current
+     * data
+     */
+    @Nullable
+    public PrivateKey getPrivateKey() {
+        if (pemObject instanceof PrivateKey) {
+            return (PrivateKey) pemObject;
+        } else if (pemObject instanceof KeyPair) {
+            return ((KeyPair) pemObject).getPrivate();
+        }
+        return null;
+    }
+
+    /**
+     * Obtains raw JCA or BouncyCastle {@link Object} from the read PEM. Depending on the PEM nature or the object
+     * passed to the {@link #PEMManager(Object pemObject)}, the returned object can be one of the following (not
+     * exhaustive list) and any classes that inherit from them:
+     * <ul>
+     * <li><strong>JCA</strong>
+     * <ul>
+     * <li>{@link Certificate}
+     * <li>{@link java.security.cert.CRL}
+     * <li>{@link KeyPair}
+     * <li>{@link PublicKey}
+     * <li>{@link PrivateKey}
+     * </ul>
+     * </ul>
+     * <ul>
+     * <li><strong>Bouncy Castle</strong>
+     * <ul>
+     * <li>{@link org.bouncycastle.asn1.cms.ContentInfo}
+     * <li>{@link org.bouncycastle.jce.spec.ECNamedCurveParameterSpec}
+     * <li>{@link org.bouncycastle.pkcs.PKCS10CertificationRequest}
+     * <li>{@link org.bouncycastle.jce.provider.X509CertificateObject}
+     * <li>{@link org.bouncycastle.x509.X509V2AttributeCertificate}
+     * </ul>
+     * </ul>
+     * 
+     * @return {@link Object} read from the PEM
+     */
+    @Nullable
```

Comment:
We rely on an external library that does the enconding and there are some parsers that in some cases return null. I can't ensure that it will be `Notnull`.